docs/HowToUseTaskMonitor.md

```diff
@@ -0,0 +1,79 @@
+It checks if registered tasks/pthreads are alive.
+
+## Contents
+- [Enable task-monitor](#enable-task-monitor)
+- [Use task-monitor](#use-task-monitor)
+
+
+## Enable task-monitor
+To enable task-monitor, please find four steps as shown below:
+
+## 1. Enable task monitor
+
+In menuconfig, enable **CONFIG_TASK_MONITOR**.
+	```
+	Task Monitor -> Enable Task Monitor
+	```
+
+## 2. Set priority of task monitor
+
+In menuconfig, Set **CONFIG_TASK_MONITOR_PRIORITY** value which indicates priority of task monitor kernel thread.  
+	```
+	Task monitor -> Priority of Task Monitor
+	```
+
+
+## 3. Set interval for checking alive(sec)
+
+In menuconfig, Set **CONFIG_TASK_MONITOR_INTERVAL** value which indicates interval for checking alive(sec)
+	```
+	Task monitor -> Interval for checking alive(sec) -> change a number over 0
+	```
+	
+Task Monitor checks the registered tasks/pthreads active status every this interval. If TASK_MONITOR_INTERVAL set by 5, Interval value can use as 5, 10, 15 .... and TASK_MONITOR_MAX_INTERVAL.
+	
+## 4. Set max interval for checking alive(sec)
+
+In menuconfig, Set **CONFIG_TASK_MONITOR_MAX_INTERVAL** value which indicates max interval for checking alive(sec)
+	```
+	Task monitor -> Max interval for checking alive(sec) -> change a number over CONFIG_TASK_MONITOR_INTERVAL
+	```
+
+It is maximum interval for task check and should be set to a multiple of TASK_MONITOR_INTERVAL. If it is not a multiple of TASK_MONITOR_INTERVAL, it is set to the previous multiple.
+ex) If TASK_MONITOR_INTERVAL set by 5 and TASK_MONITOR_MAX_INTERVAL set by 17, Interval value can use as 5, 10 and 15.
+## Use task-monitor
+To use task-monitor, please find a link as shown below:  
+[Supported APIs](#supported-apis)
+
+## Example of Task-monitor
+If you want to reset the board in some special case for example deadlock, then you can use task-monitor as shown below.
+```
+#include <tinyara/task_monitor.h>
+#define INTERVAL 5
+sem_t sem; // Assume that it is initialized.
+int main
+{
+    task_monitor_register(INTERVAL);
+    while(true) {
+        sem_wait(&sem);
+        task_monitor_update_status();
+        sem_post(&sem);
+    }
+}
+```
+
+## Supported APIs
+A header file [task_monitor.h](../os/include/tinyara/task_monitor.h) provides following APIs which support to task-monitor management as shown below:
+```
+int task_monitor_register(int interval);
+void task_monitor_update_status(void);
+```
+user can call **task_monitor_register(int interval)** function to register specific tasks/pthreads into task-monitor kernel thread with specific interval.
+
+The task monitor kernel thread checks the active state of the process at the interval initially registered, changes it to false if true, and resets the board if false.
+
+user can call the **task_monitor_update_status(void)** function to update the active state of the process to true so that task-monitor kernel thread does not terminate registered process even after the interval initially registered.
```

Comment:
1. user -> User
2. In TizenRT, there is no process notation. Please change it to task or thread.
2. User doesn't need to know the status is changed to true. And Task Monitor does not terminate the registered task.(It will reset the board.)

Reply:
modified